engine/src/flutter/lib/ui/painting.dart

```diff
@@ -2551,17 +2551,13 @@ Future<Codec> instantiateImageCodecWithSize(
 }) async {
   getTargetSize ??= _getDefaultImageSize;
   final ImageDescriptor descriptor = await ImageDescriptor.encoded(buffer);
-  try {
-    final TargetImageSize targetSize = getTargetSize(descriptor.width, descriptor.height);
-    assert(targetSize.width == null || targetSize.width! > 0);
-    assert(targetSize.height == null || targetSize.height! > 0);
-    return descriptor.instantiateCodec(
-      targetWidth: targetSize.width,
-      targetHeight: targetSize.height,
-    );
-  } finally {
-    buffer.dispose();
```

Comment:
Why was this moved from `instantiateImageCodecWithSize`?

`instantiateImageCodecWithSize` is still a public API, so now we've introduced a breaking change (memory leak) where previously calls to `instantiateImageCodecWithSize` used to dispose the buffer (in fact the documentation above) still states it:

```dart
/// The [buffer] will be disposed by this method once the codec has been created.
```

... and now it won't.

Reply:
It would be better to revert this change so as not to create breaking change.
And it would be good to have the first PR and the fix that calls `ImageDescriptor.dispose` reviewed again.

I considered it necessary to make this change as the review had stopped after I submitted my PR. However, it appears to be unnecessary. My aim is to fix https://github.com/flutter/flutter/issues/150016 and avoid the leak by calling `ImageDescriptor.dispose`. For this reason, I have no objection to revert [02c9342](https://github.com/flutter/flutter/pull/161488/commits/02c9342c72ce408c50600121eb710a8a1a8478be).